In [ ]:
#import
from datetime import datetime as dt
import pandas as pd
# 1. Gender Sales Analysis
df = pd.read_csv('Retail Sales Dataset.csv')
gender_more_buy = df.groupby('Gender')['Total Amount'].sum()
pergentage_gender = (gender_more_buy.iloc[0]) / (gender_more_buy.iloc[0] + gender_more_buy.iloc[1]) * 100
print(gender_more_buy.iloc[0]),
print(gender_more_buy.iloc[1]),
print( pergentage_gender)


242950
233210
51.02276545698925


Female sales are slightly higher than male sales, but the distribution is very close, showing a balanced customer base between genders (Females 51% vs. Males 49%).

In [9]:
## 2. Monthly Sales Trend Analysis
df['Date'] = pd.to_datetime(df['Date'], format='%d/%m/%Y')
df['month_benefits'] = df['Date'].dt.month
df['year_benefits'] = df['Date'].dt.year
month_year_sales= df.groupby(['year_benefits', 'month_benefits'])['Total Amount'].sum()
print(month_year_sales)

year_benefits  month_benefits
2023           1                 38630
               2                 45410
               3                 30330
               4                 38480
               5                 56450
               6                 36715
               7                 35865
               8                 37260
               9                 23620
               10                50080
               11                37100
               12                44690
2024           1                  1530
Name: Total Amount, dtype: int64


Peak Month: May 2023 recorded the highest sales reaching $56,450.

Lowest Month (2023): September recorded the lowest sales in 2023 at $23,620.

Note on 2024 Data: Sales for January 2024 appear very low ($1,530) due to incomplete data recording for that year in the dataset

In [ ]:
## 3. Age Analysis per Category
mean_age_product_gategory = df.groupby('Product Category')['Age'].mean()
print(mean_age_product_gategory)

Product Category
Beauty         40.371336
Clothing       41.948718
Electronics    41.736842
Name: Age, dtype: float64


The average age across all product categories is nearly identical (around 40 to 41 years old). This indicates that the products aren't targeting a specific age group, and the store's customer base is fairly balanced.

In [12]:
more_quant = df.groupby('Product Category')['Quantity'].sum()
print(more_quant)

Product Category
Beauty         811
Clothing       939
Electronics    882
Name: Quantity, dtype: int64


Clothing is the top-performing category in terms of quantity sold with 939 units, followed by Electronics with 882 units, while Beauty came in last with 811 units.

In [13]:
more_city = df.groupby('City')['Total Price'].sum()
print(more_city)

City
Atlanta          38345
Boston           45455
Chicago          49660
Dallas           47965
Houston          46085
Los Angeles      39470
Miami            54210
New York         34390
San Francisco    71335
Seattle          49245
Name: Total Price, dtype: int64


San Francisco is by far the top-performing city in overall revenue, crossing $71,000. On the other hand, New York is lagging behind as the lowest-selling city (around $34,000), which might need better marketing efforts or operational reviews.

## Executive Summary
Key takeaways from the analysis:
1. Sales are almost evenly split between genders (Females 51% vs. Males 49%).
2. May 2023 was the peak sales month, while September 2023 was the lowest.
3. Clothing is the best-selling category by volume (units sold).
4. San Francisco is the strongest market, while New York needs performance improvement.